src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricEventSourceTests.cs

```diff
@@ -2313,12 +2314,24 @@ void WaitForEvent(TimeSpan timeout, int numEvents, string eventName)
                     return;
                 }
                 TimeSpan remainingTime = stopTime - DateTime.Now;
-                if (remainingTime.TotalMilliseconds < 0 || !_autoResetEvent.WaitOne(remainingTime))
+                if (remainingTime.TotalMilliseconds < 0)
                 {
                     int currentEventCount = GetCountEvents(eventName);
                     throw new TimeoutException($"Timed out waiting for a {eventName} event. " +
                         $"StartTime={startTime} stopTime={stopTime} initialEventCount={initialEventCount} currentEventCount={currentEventCount} targetEventCount={numEvents}");
                 }
+#if OS_ISBROWSER_SUPPORT
+                if (OperatingSystem.IsBrowser())
+                {
+                    // in the single-threaded browser environment, we need to yield to the browser to allow the event to be processed
+                    // we also can't block with WaitOne
+                    await Task.Delay(10);
```

Comment:
```suggestion
                    await Task.Yield();
```
Or is there a reason to delay 10ms

Reply:
This is `OuterLoop` test and we are inside `while` loop which is waiting. The propagation is on 100ms timer. So we wait up to 10 ticks. We could possibly use some non-blocking signal, like `Task` to make it better. But it's not worth the effort.

Reply:
Sounds good.

The other thing that I was going to suggest is using a TCS. We do that in tests when we want to make sure they aren't flaky (as you get to control when it gets triggered)